subxt/src/utils/mod.rs

```diff
@@ -47,6 +50,21 @@ pub(crate) fn strip_compact_prefix(bytes: &[u8]) -> Result<(u64, &[u8]), codec::
     Ok((val.0, *cursor))
 }
 
+/// A URL is considered secure if it uses a secure scheme ("https" or "wss") or is referring to localhost.
+///
+/// Returns an error if the the string could not be parsed into a URL.
+pub fn url_is_secure(url: &str) -> Result<bool, Error> {
+    let url = Url::parse(url).map_err(|e| Error::Rpc(RpcError::ClientError(Box::new(e))))?;
+
+    let secure_scheme = url.scheme() == "https" || url.scheme() == "wss";
+    let is_localhost = url
+        .host_str()
```

Comment:
It may be better to use https://docs.rs/url/latest/url/struct.Url.html#method.host instead.
You could then match whether it's a ipv6, ipv4 or domain, https://docs.rs/url/latest/url/enum.Host.html, then the nice thing is that the Ipv6/Ipv4 types exposes `fn is_loopback()`

This is fine though and I don't think we care about ipv6 addresses anyway.